src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.cpp

```diff
@@ -40,34 +40,70 @@
 #include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.h>
 #include <OpenMS/MATH/STATISTICS/StatisticFunctions.h>
 
-// #define DEBUG
+//#define DEBUG
 
 using namespace std;
 using namespace boost::math;
 
 namespace OpenMS
 {
 
-  MultiplexFilteringProfile::MultiplexFilteringProfile(MSExperiment& exp_profile, const MSExperiment& exp_picked, const std::vector<std::vector<PeakPickerHiRes::PeakBoundary> >& boundaries, const std::vector<MultiplexIsotopicPeakPattern>& patterns, int isotopes_per_peptide_min, int isotopes_per_peptide_max, double intensity_cutoff, double rt_band, double mz_tolerance, bool mz_tolerance_unit, double peptide_similarity, double averagine_similarity, double averagine_similarity_scaling, String averagine_type) :
-    MultiplexFiltering(exp_picked, patterns, isotopes_per_peptide_min, isotopes_per_peptide_max, intensity_cutoff, rt_band, mz_tolerance, mz_tolerance_unit, peptide_similarity, averagine_similarity, averagine_similarity_scaling, averagine_type), boundaries_(boundaries)
+  MultiplexFilteringProfile::MultiplexFilteringProfile(MSExperiment& exp_profile, const MSExperiment& exp_centroided, const std::vector<std::vector<PeakPickerHiRes::PeakBoundary> >& boundaries, const std::vector<MultiplexIsotopicPeakPattern>& patterns, int isotopes_per_peptide_min, int isotopes_per_peptide_max, double intensity_cutoff, double rt_band, double mz_tolerance, bool mz_tolerance_unit, double peptide_similarity, double averagine_similarity, double averagine_similarity_scaling, String averagine_type) :
+    MultiplexFiltering(exp_centroided, patterns, isotopes_per_peptide_min, isotopes_per_peptide_max, intensity_cutoff, rt_band, mz_tolerance, mz_tolerance_unit, peptide_similarity, averagine_similarity, averagine_similarity_scaling, averagine_type)
   {
+    // initialise peak boundaries
+    // In the MultiplexFiltering() constructor we initialise the centroided experiment exp_centroided_.
+    // (We run a simple intensity filter. Peaks below the intensity cutoff can be discarded right from the start.)
+    // Now we still need to discard boundaries of low intensity peaks, in order to preserve the one-to-one mapping between peaks and boundaries.
     
-    if (exp_profile.size() != exp_picked.size())
+    // loop over spectra and boundaries
+    MSExperiment::ConstIterator it_rt;
+    std::vector<std::vector<PeakPickerHiRes::PeakBoundary> >::const_iterator it_rt_boundaries;
+    for (it_rt = exp_centroided.begin(), it_rt_boundaries = boundaries.begin();
+         it_rt < exp_centroided.end(), it_rt_boundaries < boundaries.end();
+         ++it_rt, ++it_rt_boundaries)
```

Comment:
I suggest

```
for (auto it_rt = exp_centroided.begin(), auto it_rt_boundaries = boundaries.begin();
         it_rt != exp_centroided.end() && it_rt_boundaries != boundaries.end();
        ++it_rt, ++it_rt_boundaries)
```

I dont think you want the comma operator here:

```
In a comma expression E1, E2, the expression E1 is evaluated, its result is discarded
```

Reply:
Cannot get the double auto loop to work. Complains about second auto.
```
/Users/lars/Code/OpenMS-develop/OpenMS/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.cpp:62:47: error: expected
      unqualified-id
    for (auto it_rt = exp_centroided.begin(), auto it_rt_boundaries = boundaries.begin();
```

Reply:
please don't use the comma operator in for loops. And it's not about the auto...

for (int i = 0, int b = 0; i != 10; ++i)
{
}
will also not compile.

Reply:
Ok. I will loop over one container and access the other two via indices.